server/odc-service/src/main/java/com/oceanbase/odc/service/diagnose/fulllinktrace/JaegerConverter.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.oceanbase.odc.service.diagnose.fulllinktrace;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.CollectionUtils;
+
+import com.google.gson.GsonBuilder;
+import com.oceanbase.odc.core.shared.model.TraceSpan;
+
+import lombok.Builder;
+import lombok.Getter;
+
+public class JaegerConverter implements ThirdPartyTraceConverter {
+    private static final String PROCESS_ID_PATTERN = "%s-%s-%s";
+
+    @Override
+    public String convert(TraceSpan span) {
+        Map<String, Object> dataMap = new HashMap<>();
+        dataMap.put("traceID", span.getLogTraceId());
+        List<Span> spans = new ArrayList<>();
+        Map<String, Process> processes = new HashMap<>();
+        dataMap.put("spans", spans);
+        dataMap.put("processes", processes);
+
+        dfs(spans, processes, span, null);
+
+        Map<String, Object> root = new HashMap<>();
+        root.put("data", Collections.singletonList(dataMap));
+        return new GsonBuilder().setPrettyPrinting().create().toJson(root);
```

Comment:
ODC use Jackson, avoid Gson if not neccesary

Reply:
The reason for using Gson here is to beautify. Gson can output tree-shaped JSON files, while jackson can only be tiled.
Do I need to change this?

Reply:
done, used Jackson instead of Gson